### PR TITLE
Sundering and a lot of gun changes.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
This adds shred/sundering values to about most marine guns, and changes the base recovery to 0.5. The idea is to allow marine attrition over longer periods of time.
Values are subject to change. 
I adopted most of the PFC changes after consulting with Kuro from #3774, he's adding the xeno buffs soon in a separate PR.
I buffed the MBX because it was underperforming heavily.
## Why It's Good For The Game

Adds sundering as a major thing.
Sundering is a good mechanic that allows marines to be more offensive in a META all about stalling.
Some minor buffs/nerfs to other guns. Pistols lost movement in-accuracy because that seems to be deprecated and misleading.
Lasgun got some buffs. It's shred heavy now.
SMG bullet went from 23/0 to 20/5 and now has 0.15 FD.
I gave shred to just about everything in the game.
Carbine got nerfed too. Forgot.
MBX was underperforming heavily! I tried to give it a niche as a fast boi shotgun. Let's see how it goes.
DMR does 70/20 now, and slugs do 60/40.
~~time to get crucified.~~
## Changelog
:cl:
balance: Too much to list. But I added Sunder to most weapons in the game. If not all! Also MBX buffs. Checkout https://github.com/tgstation/TerraGov-Marine-Corps/pull/3818 for more detailed changes.
/:cl:

### CODE REVIEW

**Comment by terranaut1** [comment]
Have you done any calculations about how huge of a DPS increase just a few sundering stacks are? It's not very noticeable on already squishy castes but tanky castes will really, really quickly lose the tankiness they are supposed to have and die like everything else. I'll not say it's not an interesting mechanic, it is, but it's really awful for existing balance and you can't implement this without adjusting xeno stats (and please don't say "they just were", that was for unrelated reasons and things are still/again fucked)

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1075,6 +1126,7 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	damage = 85 // this is gotta hurt...
 	max_range = 40
 	penetration = 100
+	sundering = 100
```
Comment (psykzz) [inline code/modules/projectiles/ammo_datums.dm:1129]: this will instantly remove **all** armor from whatever xeno is hit.

Comment (Pariah919) [reply code/modules/projectiles/ammo_datums.dm:1129]: I know that! I've tested it! It's a DEATHSQUAD gun! It's already penetrating fully. It's more of a showing to the player that they should run. Really really FAST.

**Comment by Pariah919** [comment]
DO NOT TM THIS UNTIL KURO RE-DOES THE XENOMORPH CASTE REBALANCE
IT WILL NOT END WELL.

**Comment by Pariah919** [comment]
PR is done. #3831. TM with that.

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -801,15 +840,17 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 	damage = 20
 	scatter = -10
 	penetration = 20
+	sundering = 1.5
 
 /datum/ammo/bullet/smartgun/lethal
-	flags_ammo_behavior = AMMO_BALLISTIC
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	icon_state 	= "bullet"
+	sundering = 5 // boolet go fasterr
```
Comment (factoryman942) [inline code/modules/projectiles/ammo_datums.dm:853]: no don't

File: code/modules/projectiles/ammo_datums.dm [hunk 3]
```diff
@@ -459,7 +478,7 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	accurate_range_min = 0
 	accurate_range = 30
-	damage = 55
+	damage = 70
```
Comment (factoryman942) [inline code/modules/projectiles/ammo_datums.dm:481]: big undocumented change

File: code/modules/projectiles/ammo_datums.dm [hunk 4]
```diff
@@ -490,8 +509,8 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	shell_speed = 3
 	max_range = 15
-	damage = 50
-	penetration = 10
+	damage = 80
+	penetration = 40
```
Comment (factoryman942) [inline code/modules/projectiles/ammo_datums.dm:513]: bigger undocumented change

File: code/modules/projectiles/ammo_datums.dm [hunk 5]
```diff
@@ -541,30 +561,35 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 	hud_state = "shotgun_flechette"
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	bonus_projectiles_type = /datum/ammo/bullet/shotgun/flechette_spread
 	bonus_projectiles_amount = 2
 	bonus_projectiles_scatter = 3
 	accuracy_var_low = 8
 	accuracy_var_high = 8
 	max_range = 15
-	damage = 75
+	damage = 50
```
Comment (factoryman942) [inline code/modules/projectiles/ammo_datums.dm:571]: undocumented

File: code/modules/projectiles/ammo_datums.dm [hunk 6]
```diff
@@ -541,30 +561,35 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
...
 
 /datum/ammo/bullet/shotgun/flechette_spread
 	name = "additional flechette"
 	icon_state = "flechette"
+	flags_ammo_behavior = AMMO_BALLISTIC|AMMO_SUNDERING
 	accuracy_var_low = 8
 	accuracy_var_high = 8
 	max_range = 15
-	damage = 30
+	damage = 50
```
Comment (factoryman942) [inline code/modules/projectiles/ammo_datums.dm:583]: undocumented

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -77,8 +77,8 @@
     starting_attachment_types = list(/obj/item/attachable/stock/t18stock)
     attachable_offset = list("muzzle_x" = 30, "muzzle_y" = 16,"rail_x" = 10, "rail_y" = 19, "under_x" = 18, "under_y" = 13, "stock_x" = 0, "stock_y" = 13)
 
-    fire_delay = 0.25 SECONDS
-    burst_delay = 0.10 SECONDS
+    fire_delay = 0.3 SECONDS
+    burst_delay = 0.125 SECONDS
```
Comment (factoryman942) [inline code/modules/projectiles/guns/rifles.dm:81]: undocumented

File: code/modules/projectiles/guns/rifles.dm [hunk 8]
```diff
@@ -156,7 +156,6 @@
 	max_shells = 10 //codex
 	current_mag = /obj/item/ammo_magazine/rifle/standard_dmr
 	attachable_allowed = list(
-						/obj/item/attachable/quickfire,
```
Comment (factoryman942) [inline code/modules/projectiles/guns/rifles.dm:159]: "

File: code/modules/projectiles/guns/rifles.dm [hunk 9]
```diff
@@ -177,7 +176,7 @@
 						/obj/item/attachable/scope,
 						/obj/item/attachable/scope/mini)
 
-	flags_gun_features = GUN_AUTO_EJECTOR|GUN_AMMO_COUNTER|GUN_LOAD_INTO_CHAMBER
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_AMMO_COUNTER|GUN_LOAD_INTO_CHAMBER|GUN_CAN_POINTBLANK
```
Comment (factoryman942) [inline code/modules/projectiles/guns/rifles.dm:179]: "

**Comment by Rohesie** [comment]
Looks like both marines and xenos are being buffed. Rocket tag times?

**Comment by Mr-Worth** [comment]
From observing as admin and playing xeno a bit I don't really think marines need a weapons buff. They already deal huge amounts of damage to all castes. A lone David should not be able to take on 3 xenos at once with just a shotgun or single marine walking in with a LMG on full auto being able to make the whole hive run away.

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-1). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by terranaut1** [comment]
Why does a PR with its only comments being concerns about balance get merged after it is made even stronger with no further balance adjustments?

**Pull request merged**